src/highdicom/sr/templates.py

```diff
@@ -247,19 +255,68 @@ def _get_roi_reference_type(
 
     Returns
     -------
-    pydicom.sr.coding.Code
-        The reference_type of the measurement group as a coded concept. None
-        if no reference type is found.
+    highdicom.sr.CodedConcept
+        Coded concept representing the concept name of the ROI reference item.
+    List[pydicom.dataset.Dataset]
+        Datasets representing the content item that defines the reference to the
+        ROI.
+
+    Raises
+    ------
+    RuntimeError:
+        If no content item representing a valid content type is found. If
+        multiple valid content items are found with different concept names.
 
     """
+    ref_type_value_type_map = {
+        codes.DCM.ImageRegion: [
+            ValueTypeValues.SCOORD,
+            ValueTypeValues.SCOORD3D
+        ],
+        codes.DCM.VolumeSurface: [ValueTypeValues.SCOORD3D],
+        codes.DCM.ReferencedSegment: [ValueTypeValues.IMAGE],
+        codes.DCM.ReferencedSegmentationFrame: [ValueTypeValues.IMAGE],
+        _REGION_IN_SPACE: [ValueTypeValues.COMPOSITE],
+    }
+    returned_items = []
+    reference_type = None
     for item in group_item.ContentSequence:
-        for concept_name in allowed_reference_types:
-            found_concept = CodedConcept.from_dataset(
-                item.ConceptNameCodeSequence[0]
-            )
-            if found_concept == concept_name:
-                return concept_name
-    return None
+        rel_type = RelationshipTypeValues(item.RelationshipType)
+        if rel_type != RelationshipTypeValues.CONTAINS:
```

Comment:
Could we now directly use `item.relationship_type`?

```suggestion
        if item.relationship_type != RelationshipTypeValues.CONTAINS:
```

Reply:
Resolved in bd0f5d3